pkg/codegen/operations.go

```diff
@@ -262,6 +264,62 @@ func (o *OperationDefinition) SummaryAsComment() string {
 	return strings.Join(parts, "\n")
 }
 
+// Produces a list of type definitions for a given Operation for the response
+// types which we know how to parse. These will be turned into fields on a
+// response object for automatic deserialization of responses in the generated
+// Client code. See "client-with-responses.tmpl".
+// This produces one type definition for each JSON response body.
+func (o *OperationDefinition) GetResponseIndependentTypeDefinitions() ([]TypeDefinition, error) {
```

Comment:
What was the motivation behind this? Looking at the diff ([generated.go.diff.txt](https://github.com/curvegrid/oapi-codegen/files/5156062/generated.go.diff.txt)) it seems that it will make the use of the generated code slightly more boilerplate no?

Reply:
from @natsukagami:
> before that change I had all response types aliased (for ctx response methods), as they are not generated by default. However aliasing types cannot have validation, which is required for testing, so I had to generate actual response types in the server template too.
> The response types in the original (client with reaponses, with the TypedClient switch  i added) also generate these types, but they are inlined in a big "Response" struct